airbyte-integrations/bases/source-acceptance-test/unit_tests/test_incremental.py

```diff
@@ -65,16 +85,61 @@ def test_compare_cursor_with_threshold(record_value, state_value, threshold_days
 @pytest.mark.parametrize(
     "records1, records2, latest_state, threshold_days, expected_error",
     [
-        ([{"date": "2020-01-01"}, {"date": "2020-01-02"}], [], "2020-01-02", 0, None),
-        ([{"date": "2020-01-02"}, {"date": "2020-01-03"}], [], "2020-01-02", 0, "First incremental sync should produce records younger"),
-        ([{"date": "2020-01-01"}, {"date": "2020-01-02"}], [{"date": "2020-01-02"}, {"date": "2020-01-03"}], "2020-01-02", 0, None),
-        ([{"date": "2020-01-01"}], [{"date": "2020-01-01"}], "2020-01-02", 0, "Second incremental sync should produce records older"),
-        ([{"date": "2020-01-01"}, {"date": "2020-01-02"}], [{"date": "2020-01-01"}, {"date": "2020-01-02"}], "2020-01-03", 2, None),
-        ([{"date": "2020-01-02"}, {"date": "2020-01-03"}], [], "2020-01-02", 2, "First incremental sync should produce records younger"),
-        ([{"date": "2020-01-01"}], [{"date": "2020-01-02"}], "2020-01-06", 3, "Second incremental sync should produce records older"),
+        ([{"date": "2020-01-01"}, {"date": "2020-01-02"}], [], "2020-01-02", 0, does_not_raise()),
+        (
+            [{"date": "2020-01-02"}, {"date": "2020-01-03"}],
+            [],
+            "2020-01-02",
+            0,
+            pytest.raises(AssertionError, match="First incremental sync should produce records younger"),
```

Comment:
```suggestion
            pytest.raises(AssertionError, match="First incremental sync should produce records older"),
```
I think the original error statement should be the opposite. This test fails because the sync produced records younger than the state value.

Reply:
tracked in https://github.com/airbytehq/airbyte/issues/16894